packages/rum/src/domain/record/observers/viewportResizeObserver.ts

```diff
@@ -13,22 +12,25 @@ export type ViewportResizeCallback = (incrementalSnapshotRecord: BrowserIncremen
 
 export type VisualViewportResizeCallback = (visualViewportRecord: VisualViewportRecord) => void
 
-export function initViewportResizeObserver(
-  configuration: RumConfiguration,
-  viewportResizeCb: ViewportResizeCallback
-): ListenerHandler {
-  return initViewportObservable(configuration).subscribe((data: ViewportDimension) => {
+export function initViewportResizeObserver(configuration: RumConfiguration, viewportResizeCb: ViewportResizeCallback) {
+  const viewportResizeSubscription = initViewportObservable(configuration).subscribe((data: ViewportDimension) => {
     viewportResizeCb(assembleIncrementalSnapshot<ViewportResizeData>(IncrementalSource.ViewportResize, data))
-  }).unsubscribe
+  })
+
+  return {
+    stop: () => {
+      viewportResizeSubscription.unsubscribe()
+    },
+  }
 }
 
 export function initVisualViewportResizeObserver(
   configuration: RumConfiguration,
   visualViewportResizeCb: VisualViewportResizeCallback
-): ListenerHandler {
+) {
   const visualViewport = window.visualViewport
   if (!visualViewport) {
-    return noop
+    return { stop: () => {} }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ 

```suggestion
    return { stop: noop }
```